security/pkg/pki/ca/ca.go

```diff
@@ -405,3 +405,10 @@ func (ca *IstioCA) signWithCertChain(csrPEM []byte, subjectIDs []string, request
 	}
 	return cert, nil
 }
+
+// Adds Callback for when the CA root changes
+func (ca *IstioCA) AddRootCertCb(cb func(string) error) {
```

Comment:
Can we revert the changes on this and selfsignedcarootcertrotator.go?
I don't think we need to dig so much. Consuming Citadel root via the root config is only during the migration, and a rotation normally happens at most once per year.